diff --git a/go/private/rules/cgo.bzl b/go/private/rules/cgo.bzl
index 6fe41447..ab8adaab 100644
--- a/go/private/rules/cgo.bzl
+++ b/go/private/rules/cgo.bzl
@@ -161,7 +161,9 @@ def cgo_configure(go, srcs, cdeps, cppopts, copts, cxxopts, clinkopts):
                     clinkopts.extend(["-L", lib.dirname, "-l", libname])
                     inputs_direct.append(lib)
                 else:
+                    lib_opts.append("-Wl,--whole-archive")
                     lib_opts.append(lib.path)
+                    lib_opts.append("-Wl,--no-whole-archive")
             clinkopts.extend(cc_link_flags(d))
 
         elif hasattr(d, "objc"):
